javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -219,5 +219,100 @@ object Math {
     }
   }
 
-  // TODO The methods not available in the JavaScript Math object
+  def addExact(a: scala.Int, b: scala.Int): scala.Int = {
+    val overflow =
+      b > 0 && a > Integer.MAX_VALUE - b || b < 0 && a < Integer.MIN_VALUE - b
+    if (!overflow) a + b
+    else throw new ArithmeticException("Integer overflow")
+  }
+
+  def addExact(a: scala.Long, b: scala.Long): scala.Long = {
+    val overflow =
+      b > 0 && a > Long.MAX_VALUE - b || b < 0 && a < Long.MIN_VALUE - b
+    if (!overflow) a + b
+    else throw new ArithmeticException("Long overflow")
```

Comment:
You can minimize the number of operations on Longs by doing something like this instead:
```scala
val r = a + b
val rSign = Long.signum(r)
if (rSign == Long.signum(a) || rSign == Long.signum(b)) r
else throw new ArithmeticException("Long overflow")
```
`Long.signum` is intrinsified by the optimizer to directly work on the internals of the `Long` representation, and returns cached "instances" of -1, 0 and 1.